### PR TITLE
R2 Support

### PR DESCRIPTION
**Pull request by ASpoonPlaysGames**
This currently does completely destroy R5 support, so uhhh dont merge it yet, just making this so that it can be seen

### CODE REVIEW

**Comment by Malik4626572** [comment]
amogus

File: RePak/include/rpak.h [hunk 1]
```diff
@@ -245,47 +319,54 @@ enum class dtblcoltype_t : uint32_t
...
-	uint16_t format = 0;
+	uint16_t unk0 = 0;
+	uint16_t m_nFormat = 0;
 
-	uint32_t dataLength; // total data size across all sources
-	uint8_t unknown_2;
-	uint8_t optStreamedMipLevels; // why is this here and not below? respawn moment
+	uint32_t m_nDataLength; // total data size across all sources
+	uint8_t unk1;
+	uint8_t m_nOptSreamedMipLevels; // why is this here and not below? respawn moment
```
Comment (r-ex) [inline RePak/include/rpak.h:333]: sreamed

Comment (ASpoonPlaysGames) [reply RePak/include/rpak.h:333]: Sreamed

**Comment by r-ex** [comment]
superseded by #3

**Pull request closed without merging**